src/cheri_cap_common.sail

```diff
@@ -488,6 +488,48 @@ function setCapBounds(cap, base, length) : (Capability, CapAddrBits, CapAddrBits
   (exact, newCap)
 }
 
+/* XXX this should go in prelude.sail but putting here for now to avoid submodule faff */
+val count_trailing_zeros : forall 'N, 'N >=0 . bits('N) -> range(0, 'N)
+function count_trailing_zeros(bv) = {
+  foreach(i from 0 to ('N - 1) by 1 in inc) {
+    if bv[i] == bitone then {
+      return i;
+    }
+  };
+  'N
+}
+
+/*!
+ * Returns cap the address and base set to the given base (exactly) and the
+ * length set to the largest value less than or equal to given length that can
+ * be represented while preserving the alignment of the base.
+ */
+function setCapBoundsRoundDown(cap, base, length) : (Capability, CapAddrBits, CapAddrBits) -> Capability = {
```

Comment:
We should probably include this function in the arch doc in Appendix B as we do for setCapBounds?